spring-5-reactive-client/src/test/java/com/baeldung/reactive/service/ReactiveUploadServiceUnitTest.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.reactive.service;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ReactiveUploadServiceUnitTest {
+
+    private static final String BASE_URL = "http://localhost:8080/external/upload";
+
+    final WebClient webClientMock = WebClient.builder().baseUrl(BASE_URL)
+            .exchangeFunction(clientRequest -> Mono.just(ClientResponse.create(HttpStatus.OK)
+               .header("content-type", "application/json")
+               .build()))
+            .build();
+
+    private final ReactiveUploadService tested = new ReactiveUploadService(webClientMock);
+
+    @Test
+    void givenAPdf_whenUploadingWithWebClient_thenOK() {
+        final Resource file = mock(Resource.class);
+        final Mono<HttpStatus> result = tested.uploadPdf(file);
```

Comment:
Put blank line between given/when and when/then parts of each test. In this example, this would be done before calling the upload method and before the assertion.